src/Raven.Server/Smuggler/Documents/DatabaseSource.cs

```diff
@@ -108,6 +109,12 @@ public Task<SmugglerInitializeResult> InitializeAsync(DatabaseSmugglerOptionsSer
                 {
                     LastRaftIndex = rawRecord.EtagForBackup;
                 }
+
+                if (options.IgnoreCorruptedDocumentsErrors)
+                {
+                    _database.DocumentsStorage.OnCorruptedDocumentHandler = OnCorruptedDocument;
+                    _result = result;
```

Comment:
this is racy, because you are assigning handler first and then this result and the event might trigger in-between

Reply:
Corrected, as now the handler is located inside `result`, and the export process occurs within a single thread.